processor/src/errors.rs

```diff
@@ -21,10 +26,95 @@ pub enum ExecutionError {
-    InterpolationResultSizeTooBig(u64),
+    InterpolationResultSizeTooBig(usize, usize),
     NotBinaryValue(Felt),
     NotU32Value(Felt),
     ProverError(ProverError),
     SyscallTargetNotInKernel(Digest),
-    UnexecutableCodeBlock(CodeBlock),
+    UnexecutableProxyBlock(CodeBlock),
 }
+
+impl Display for ExecutionError {
+    fn fmt(&self, fmt: &mut Formatter<'_>) -> Result<(), core::fmt::Error> {
+        match self {
+            ExecutionError::AdviceKeyNotFound(key) => {
+                write!(fmt, "Value {key:?} not presented on the advice map, can't write it to tape")
+            }
+            ExecutionError::AdviceSetLookupFailed(reason) => {
+                write!(fmt, "Advice set lookup failed, reason {reason}")
+            }
+            ExecutionError::AdviceSetNotFound(name) => {
+                write!(fmt, "Advice set {name:?} not found")
+            }
+            ExecutionError::AdviceSetUpdateFailed(reason) => {
+                write!(fmt, "Advice set update failed, reason {reason}")
+            }
+            ExecutionError::AdviceTapeReadFailed(step) => {
+                write!(fmt, "Advice tape read failed at step {step}")
+            }
+            ExecutionError::CodeBlockNotFound(digest) => {
+                write!(
+                    fmt,
+                    "Execution of call instruction failed, function body {digest:?} not found"
+                )
+            }
+            ExecutionError::CallerNotInSyscall => {
+                write!(
+                    fmt,
+                    "Instruction `caller` used outside of kernel context, this is not supported"
+                )
+            }
+            ExecutionError::DivideByZero(clk) => {
+                write!(fmt, "Division by zero at clock {clk}")
+            }
+            ExecutionError::DuplicateAdviceKey(key) => {
+                write!(fmt, "Insertion into advice map failed because {key:?} already exists")
+            }
+            ExecutionError::FailedAssertion(clk) => {
+                write!(fmt, "Assertion failed at clock {clk}")
+            }
+            ExecutionError::UninitializedMemoryAddress(address) => {
+                write!(fmt, "Ext2INTT referenced unintialized memory at address {address}")
+            }
+            ExecutionError::InvalidFmpValue(old, new) => {
+                write!(fmt, "Updating FMP register failed, from {old} to {new}, because {new} is outside of {FMP_MIN}..{FMP_MAX}")
+            }
+            ExecutionError::NttDomainSizeTooSmall(v) => {
+                write!(fmt, "Input domain evaluation length can not be zero {v}")
+            }
+            ExecutionError::NttDomainSizeNotPowerof2(v) => {
+                write!(fmt, "Input domain evaluation must be a power of two, got {v}")
+            }
+            ExecutionError::InvalidStackDepthOnReturn(depth) => {
+                write!(fmt, "Maximum supported stack depth when returning from call is {STACK_TOP_SIZE}, got {depth}")
```

Comment:
nit: I would change this to: "When returning from a call, stack depth must be {STACK_TOP_SIZE}, but was {depth}".